src/components/calculator.js

```diff
@@ -1,40 +1,60 @@
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
+    this.handleStateChange = this.handleStateChange.bind(this);
     this.state = {
-      result: 0,
+      total: '',
+      next: '',
+      operation: '',
     };
   }
 
+  handleStateChange(newState) {
+    if (newState.total != null) {
+      this.setState({ total: newState.total });
+    }
+
+    // if (newState.next !== null) {
+    this.setState({ next: newState.next });
+    // }
+
+    if (newState.operation != null) {
+      this.setState({ operation: newState.operation });
+    }
+  }
```

Comment:
- [ ] After completeing one operation Ex: `1+2 = 3`, I tried to do another operation ` + 9`, but it shows me the error of `invalid number`. Please fix it for a smoother calculation.